### PR TITLE
Update spack-stack-dev from spack develop as of 2024/10/31

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

Update spack-stack-dev from spack develop as of 2024/10/31. The following files had merge conflicts that I needed to resolve and that I would like the reviewers to look at (there were a few more, but those I knew enough about to resolve with confidence):
```
        both modified:   .github/workflows/unit_tests.yaml
        both modified:   var/spack/repos/builtin/packages/fontconfig/package.py
        both modified:   var/spack/repos/builtin/packages/g2/package.py
        both modified:   var/spack/repos/builtin/packages/gdal/package.py
        both modified:   var/spack/repos/builtin/packages/gmake/package.py
        both modified:   var/spack/repos/builtin/packages/harfbuzz/package.py
        both modified:   var/spack/repos/builtin/packages/jasper/package.py
        both modified:   var/spack/repos/builtin/packages/landsfcutil/package.py
        both modified:   var/spack/repos/builtin/packages/libtirpc/package.py
        both modified:   var/spack/repos/builtin/packages/libxkbcommon/package.py
        both modified:   var/spack/repos/builtin/packages/openmpi/package.py
        both modified:   var/spack/repos/builtin/packages/pixman/package.py
        both modified:   var/spack/repos/builtin/packages/py-cffi/package.py
        both modified:   var/spack/repos/builtin/packages/py-metpy/package.py
        both modified:   var/spack/repos/builtin/packages/py-setuptools-rust/package.py
        both modified:   var/spack/repos/builtin/packages/qt/package.py
        both modified:   var/spack/repos/builtin/packages/sfcio/package.py
        both modified:   var/spack/repos/builtin/packages/sigio/package.py
        both modified:   var/spack/repos/builtin/packages/ufs-weather-model/package.py
```

**Update 2024/11/08** Added commit https://github.com/JCSDA/spack/pull/482/commits/a3dd052fffce42379cfe814c5550cae59539d895 to fix a bug in `cprnc` - the install rpaths for the netCDF libraries were missing, which caused the installations from the binary cache to fail during the relocation step:
```
$ ldd install/bin/cprnc
        linux-vdso.so.1 (0x00007ffde9de5000)
        libnetcdff.so.7 => not found
        libnetcdf.so.19 => not found
        libgfortran.so.5 => /home/dom/work/spack-stack/update_from_spack_dev_20241031/spack-stack-update/envs/ue-gcc-13.3.0/install/gcc/13.3.0/gcc-runtime-13.3.0-3gtnokv/lib/libgfortran.so.5 (0x0000152b8aa7d000)
        libm.so.6 => /lib64/libm.so.6 (0x0000152b8a9a2000)
        libgcc_s.so.1 => /home/dom/work/spack-stack/update_from_spack_dev_20241031/spack-stack-update/envs/ue-gcc-13.3.0/install/gcc/13.3.0/gcc-runtime-13.3.0-3gtnokv/lib/libgcc_s.so.1 (0x0000152b8a97c000)
        libc.so.6 => /lib64/libc.so.6 (0x0000152b8a773000)
        /lib64/ld-linux-x86-64.so.2 (0x0000152b8ad63000)
```

## Testing

- [x] Unit tests (GitHub actions) - note that I had to turn off the Windows Github actions tests, since they are written in a way that require the GitHub repo to be `https://github.com/spack/spack`
- [x] For testing with spack-stack, see https://github.com/JCSDA/spack-stack/pull/1363 (work in progress)

### CODE REVIEW

**Comment by AlexanderRichert-NOAA** [comment]
I'll need a bit of time to look over this. It looks good so far, though I think it'll need one or two tweaks related to static libraries...

**Review by srherbener — APPROVED** [review]
Tested on my Mac (Sonoma 14.6.1, Command Line Tools 15.3, GNU Fortran 12.4.0), and it looks good. I was able to build jedi-bundle and I got the usual ctest failures (19) that we have been seeing on the mac.

**Comment by climbfuji** [comment]
@AlexanderRichert-NOAA Please make sure to push any changes here upstream afterwards. We want the packages to be 100% identical, if possible. I will create a PR for https://github.com/JCSDA/spack/pull/482/commits/a3dd052fffce42379cfe814c5550cae59539d895 (cprnc) shortly.

**Comment by climbfuji** [comment]
spack PR for the cprnc bug fix: https://github.com/spack/spack/pull/47505

**Comment by AlexanderRichert-NOAA** [comment]
@climbfuji yep will do, I'm keeping a log, and I'll also provide a summary here when I'm done going through the list of conflicts.

**Comment by climbfuji** [comment]
Perfect, thanks a lot
________________________________

**Comment by AlexanderRichert-NOAA** [comment]
Here's my summary of the packages that were manually fixed for conflicts:
g2, gmake, harfbuzz, jasper, landsfcutil, openmpi, pixman, py-cffi, py-metpy, qt, sfcio, sigio, ufs-weather-model: good (in most cases no difference with main spack)
fontconfig: updated static deps logic
gdal: updated static deps logic for gdal, libgeotiff
libxkbcommon: updated static deps logic
py-setuptools-rust: removed duplicate py-typing-extensions dep

File: var/spack/repos/builtin/packages/libxkbcommon/package.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+    # Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
```
Comment (climbfuji) [inline var/spack/repos/builtin/packages/libxkbcommon/package.py:1]: This should be reverted, or?

**Comment by climbfuji** [comment]
> spack PR for the cprnc bug fix: [spack#47505](https://github.com/spack/spack/pull/47505)

The upstream PR for the cprnc bug fix was merged

**Pull request merged**